src/include/duckdb/common/opener_file_system.hpp

```diff
@@ -61,15 +61,15 @@ class OpenerFileSystem : public FileSystem {
 		GetFileSystem().FileSync(handle);
 	}
 
-	bool DirectoryExists(const string &directory) override {
-		return GetFileSystem().DirectoryExists(directory);
+	bool DirectoryExists(const string &directory, FileOpener *opener = nullptr) override {
+		return GetFileSystem().DirectoryExists(directory, GetOpener().get());
 	}
-	void CreateDirectory(const string &directory) override {
-		return GetFileSystem().CreateDirectory(directory);
+	void CreateDirectory(const string &directory, FileOpener *opener = nullptr) override {
+		return GetFileSystem().CreateDirectory(directory, GetOpener().get());
 	}
 
-	void RemoveDirectory(const string &directory) override {
-		return GetFileSystem().RemoveDirectory(directory);
+	void RemoveDirectory(const string &directory, FileOpener *opener = nullptr) override {
+		return GetFileSystem().RemoveDirectory(directory, GetOpener().get());
```

Comment:
Can this throw an exception if an opener is passed in, similar to the other code?

Reply:
does a `D_ASSERT` from `src/include/duckdb/common/assert.hpp` is ok instead of an exception ?

Reply:
That's fine as well